pkg/credentialprovider/plugin/plugin.go

```diff
@@ -385,12 +391,27 @@ func (e *execPlugin) ExecPlugin(ctx context.Context, image string) (*credentialp
 	cmd := exec.CommandContext(ctx, filepath.Join(e.pluginBinDir, e.name), e.args...)
 	cmd.Stdout, cmd.Stderr, cmd.Stdin = stdout, stderr, stdin
 
-	cmd.Env = []string{}
+	// Initialise envVars map with system defined environment variables
+	envVars := parseEnvVars(e.environ())
+
+	// Add all the environment variables configured in the plugin provider
+	// this will also override system env vars with the ones present in the
+	// config file giving config defined env vars a higher priority
 	for _, envVar := range e.envVars {
-		cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", envVar.Name, envVar.Value))
+		envVars[envVar.Name] = envVar.Value
+	}
+
+	cmd.Env = []string{}
```

Comment:
no need to parse into a map, just initialize `cmd.Env = e.environ()`, then append by ranging over envVars... later duplicate items in cmd.Env override